source/module_hamilt_lcao/hamilt_lcaodft/spar_hsr.cpp

```diff
@@ -25,12 +26,24 @@ void sparse_format::cal_HSR(
         hamilt::HamiltLCAO<std::complex<double>, double>* p_ham_lcao = 
         dynamic_cast<hamilt::HamiltLCAO<std::complex<double>, double>*>(p_ham);
 
-		sparse_format::cal_HContainer_d(
+        if(GlobalV::ESOLVER_TYPE == "tddft" && elecstate::H_TDDFT_pw::stype ==1)
```

Comment:
do not use GlobalV

Reply:
My apologies. I do not know the replacement for "GlobalV::ESOLVER_TYPE", I'll define a new static const bool type variable to replace it.